envoy/network/address.h

```diff
@@ -161,6 +161,7 @@ class Instance {
    * For IPv4 addresses: "1.2.3.4:80"
    * For IPv6 addresses: "[1234:5678::9]:443"
    * For pipe addresses: "/foo"
+   * For internal addresses: "serverListenerName-<hash of metadata>"
```

Comment:
Is the port directly available as a property of the Address?

Reply:
I have another API PR that also changes how binding works https://github.com/envoyproxy/envoy/pull/22274, because we don't want to bind to individual "ports". It's merely an endpoint identifier for diff tracking.

Reply:
@kyessenov will let you guide the review for API here and let me know when ready for another pass. It  seems #22274 is solving the broader problem here.